jina/jaml/__init__.py

```diff
@@ -49,17 +82,45 @@ class DummyClass:
     An expression can be any combination of literal values, references to a context, or functions.
     You can combine literals, context references, and functions using operators.
 
-    You need to use specific syntax to tell Jina to evaluate an expression rather than treat it as a string.
+    You need to use specific syntax to tell Jina to evaluate an expression rather than treat it as a string,
+    which is based on GitHub actions syntax, and looks like this:
 
     .. highlight:: yaml
     .. code-block:: yaml
 
         ${{ <expression> }}
 
-    To evaluate (i.e. substitute the value to the real value)
-    the expression when loading, use :meth:`load(substitute=True)`.
+    This expression can be evaluated directly (i.e. substituted by the real value) when being loaded,
+    by using :meth:`load(substitute=True)`
+
+    JAML supports three different kinds of variables to be used as expressions: `Environment variables`
+    (coming form the environment itself), `context variables` (being passed as a dict),
+    and `internal references` (included in the .yaml file itself).
+
+    An environment variable `var` is accessed through the following syntax:
+
+    .. highlight:: yaml
+    .. code-block:: yaml
+
+        ${{ env.var }}
+
+    Note the mandatory spaces before and after the variable denotation.
+
+    Context variables can be accessed using the following syntax:
```

Comment:
What is a context var? Isn't this the same 

```
${{ env.var }}
# and
${{ var }}
```?

Reply:
No, the latter refers exclusively to context variables, which can be passed to Flow.load_config(), Executor.load_config(), Jaml.parse() as a dict. Slightly more info is below. We didn't know about these either until we found them during testing.